From 019f6c2d838aefa9b625bd7a5eadb031fca735cd Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Mon, 27 Jan 2014 16:06:16 -0500
Subject: [PATCH] nova: add migrate.cfg to the MANIFEST

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 MANIFEST.in |    1 +
 1 file changed, 1 insertion(+)

diff --git a/MANIFEST.in b/MANIFEST.in
index c978a52dae21..61e522c610d6 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,5 +1,6 @@
 include AUTHORS
 include ChangeLog
+include nova/db/sqlalchemy/migrate_repo/migrate.cfg
 exclude .gitignore
 exclude .gitreview
 
-- 
1.7.10.4

