---
 Makefile.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- libdwarf.orig/Makefile.in
+++ libdwarf/Makefile.in
@@ -135,13 +135,13 @@ libdwarf.so: dwarf_names.h dwarf_names.c
 	$(CC) $(CFLAGS) -shared $(OBJS)  dwarf_names.o -o $@
 
 none:
 	echo "do nothing"
 common.o: $(srcdir)/common.c $(srcdir)/common.h
-	$(CC) $(CFLAGS) -c $(srcdir)/common.c
+	$(BUILD_CC) $(BUILD_CFLAGS) -c $(srcdir)/common.c
 gennames: $(srcdir)/gennames.c $(srcdir)/dwarf.h common.o
-	$(CC) $(CFLAGS) $(srcdir)/gennames.c common.o $(LDFLAGS) -o gennames
+	$(BUILD_CC) $(BUILD_CFLAGS) $(srcdir)/gennames.c common.o $(BUILD_LDFLAGS) -o gennames
 dwarf_names.c dwarf_names.h: gennames $(srcdir)/dwarf.h
 	rm -f dwarf_names.h dwarf_names.c
 	./gennames  @dwarf_namestable@  -i $(srcdir) -o .
 
 
