--- Python-2.7.9/Lib/sysconfig.py	2015-12-16 00:06:46.382635832 +0530
+++ Python-2.7.9/Lib/sysconfig.py	2015-12-16 00:24:16.266663947 +0530
@@ -7,10 +7,10 @@
 
 _INSTALL_SCHEMES = {
     'posix_prefix': {
-        'stdlib': '{base}/'+sys.lib+'/python{py_version_short}',
-        'platstdlib': '{platbase}/'+sys.lib+'/python{py_version_short}',
-        'purelib': '{base}/'+sys.lib+'/python{py_version_short}/site-packages',
-        'platlib': '{platbase}/'+sys.lib+'/python{py_version_short}/site-packages',
+        'stdlib': '{base}/lib/python{py_version_short}',
+        'platstdlib': '{platbase}/lib/python{py_version_short}',
+        'purelib': '{base}/lib/python{py_version_short}/site-packages',
+        'platlib': '{platbase}/lib/python{py_version_short}/site-packages',
         'include': '{base}/include/python{py_version_short}',
         'platinclude': '{platbase}/include/python{py_version_short}',
         'scripts': '{base}/bin',
@@ -65,10 +65,10 @@
         'data'   : '{userbase}',
         },
     'posix_user': {
-        'stdlib': '{userbase}/'+sys.lib+'/python{py_version_short}',
-        'platstdlib': '{userbase}/'+sys.lib+'/python{py_version_short}',
-        'purelib': '{userbase}/'+sys.lib+'/python{py_version_short}/site-packages',
-        'platlib': '{userbase}/'+sys.lib+'/python{py_version_short}/site-packages',
+        'stdlib': '{userbase}/lib/python{py_version_short}',
+        'platstdlib': '{userbase}/lib/python{py_version_short}',
+        'purelib': '{userbase}/lib/python{py_version_short}/site-packages',
+        'platlib': '{userbase}/lib/python{py_version_short}/site-packages',
         'include': '{userbase}/include/python{py_version_short}',
         'scripts': '{userbase}/bin',
         'data'   : '{userbase}',
