--- Python-2.7.9/Lib/site.py	2015-12-16 00:06:46.382635832 +0530
+++ Python-2.7.9/Lib/site.py	2015-12-16 00:24:16.266663947 +0530
@@ -288,15 +288,15 @@
         if sys.platform in ('os2emx', 'riscos'):
             sitepackages.append(os.path.join(prefix, "Lib", "site-packages"))
         elif os.sep == '/':
-            sitepackages.append(os.path.join(prefix, sys.lib,
+            sitepackages.append(os.path.join(prefix, "lib",
                                         "python" + sys.version[:3],
                                         "site-packages"))
-            sitepackages.append(os.path.join(prefix, sys.lib, "site-python"))
-            if sys.lib != "lib":
-                sitepackages.append(os.path.join(prefix, "lib",
-                                            "python" + sys.version[:3],
-                                            "site-packages"))
-                sitepackages.append(os.path.join(prefix, "lib", "site-python"))
+            sitepackages.append(os.path.join(prefix, "lib", "site-python"))
+            #if sys.lib != "lib":
+            #    sitepackages.append(os.path.join(prefix, "lib",
+            #                                "python" + sys.version[:3],
+            #                                "site-packages"))
+            #    sitepackages.append(os.path.join(prefix, "lib", "site-python"))
         else:
             sitepackages.append(prefix)
             sitepackages.append(os.path.join(prefix, sys.lib, "site-packages"))
