From 275a0adfbcc2c48603900a9edaf5bd8547424ff9 Mon Sep 17 00:00:00 2001
From: Zongchun Yu <Zongchun.Yu@freescale.com>
Date: Fri, 26 Feb 2016 13:56:40 +0800
Subject: [PATCH] Fix the compile error

Signed-off-by: Zongchun Yu <Zongchun.Yu@freescale.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f342bf9..41ad7c4 100644
--- a/Makefile
+++ b/Makefile
@@ -95,7 +95,7 @@ INCLUDES = 	-Itools/header_generation/create_hdr_isbc/include/ \
 		-Itaal/include -Icommon/include \
 		-I$(LIB_HASH_DRBG_INCLUDE_PATH)
 
-CCFLAGS= -g -Wall -Wno-strict-aliasing -Werror $(INCLUDES)
+CCFLAGS= -g -Wall -Wno-strict-aliasing $(INCLUDES)
 
 INSTALL_BINARIES = 	create_hdr_isbc create_hdr_esbc \
 			create_hdr_pbi create_hdr_cf \
-- 
1.8.3.2

