---
 Makefile.in |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

--- dwarfdump2.orig/Makefile.in
+++ dwarfdump2/Makefile.in
@@ -100,14 +100,17 @@ common.o:  $(srcdir)/common.cc $(srcdir)
 # in the tree builds.
 trivial_naming.o: $(srcdir)/naming.cc
 	$(CXX) $(CXXFLAGS) -DTRIVIAL_NAMING  -c $(srcdir)/naming.cc -o trivial_naming.o
 
 tag_tree_build:  $(srcdir)/tag_tree.cc $(DIRINC)/dwarf.h $(HEADERS) tag_common.o trivial_naming.o common.o
-	$(CXX) $(CXXFLAGS) $(srcdir)/tag_tree.cc trivial_naming.o tag_common.o common.o $(LDFLAGS) -o tag_tree_build 
-
+ifeq ($(NATIVE),1)
+	$(CXX) $(CXXFLAGS) $(srcdir)/tag_tree.cc trivial_naming.o tag_common.o common.o $(LDFLAGS) -o tag_tree_build
+endif
 tag_attr_build:  $(srcdir)/tag_attr.cc $(DIRINC)/dwarf.h $(HEADERS) trivial_naming.o tag_common.o naming.o common.o
-	$(CXX) $(CXXFLAGS) $(srcdir)/tag_attr.cc trivial_naming.o tag_common.o common.o $(LDFLAGS) -o tag_attr_build 
+ifeq ($(NATIVE),1)
+	$(CXX) $(CXXFLAGS) $(srcdir)/tag_attr.cc trivial_naming.o tag_common.o common.o $(LDFLAGS) -o tag_attr_build
+endif
 
 tmp-tt-table.cc tmp-tt-ext-table.cc: $(srcdir)/tag_tree_ext.list $(srcdir)/tag_tree.list tag_tree_build
 	# gcc -E tag_tree.list does not work, so use a .cc name
 	-rm -f  tmp-t1.cc
 	cp $(srcdir)/tag_tree.list tmp-t1.cc
