---
 gcc/config/aarch64/t-aarch64-linux |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gcc-linaro-4.8-2013.04.orig/gcc/config/aarch64/t-aarch64-linux
+++ gcc-linaro-4.8-2013.04/gcc/config/aarch64/t-aarch64-linux
@@ -20,6 +20,6 @@
 
 LIB1ASMSRC   = aarch64/lib1funcs.asm
 LIB1ASMFUNCS = _aarch64_sync_cache_range
 
 AARCH_BE = $(if $(findstring TARGET_BIG_ENDIAN_DEFAULT=1, $(tm_defines)),_be)
-MULTILIB_OSDIRNAMES = .=../lib64$(call if_multiarch,:aarch64$(AARCH_BE)-linux-gnu)
+MULTILIB_OSDIRNAMES = .=../lib$(call if_multiarch,:aarch64$(AARCH_BE)-linux-gnu)
