Index: qpidc-0.20/configure.ac
===================================================================
--- qpidc-0.20.orig/configure.ac
+++ qpidc-0.20/configure.ac
@@ -236,13 +236,13 @@ AM_CONDITIONAL([HAVE_PYTHON_DEVEL], [tes
 
 # Perl bindings:
 
-AC_CHECK_PROG([PERL], [perl], [perl])
-if test -n "$PERL"; then
-    PERL_ARCHLIB=`perl -MConfig -e 'print "$Config{archlib}";'`
-    AC_CHECK_FILE( ["$PERL_ARCHLIB/CORE/perl.h"],
-            [AC_SUBST([PERL_INC], ["$PERL_ARCHLIB/CORE"])] )
-    AC_SUBST(PERL_ARCHLIB)
-fi
+#AC_CHECK_PROG([PERL], [perl], [perl])
+#if test -n "$PERL"; then
+#    PERL_ARCHLIB=`perl -MConfig -e 'print "$Config{archlib}";'`
+#    AC_CHECK_FILE( ["$PERL_ARCHLIB/CORE/perl.h"],
+#            [AC_SUBST([PERL_INC], ["$PERL_ARCHLIB/CORE"])] )
+#    AC_SUBST(PERL_ARCHLIB)
+#fi
 AM_CONDITIONAL([HAVE_PERL_DEVEL], [test -n "$PERL" && test -n "$SWIG" && test -n "$PERL_INC"])
 
 specdir=`pwd`/$srcdir/../specs
