diff -urN puppet-3.7.3_ori/puppet.gemspec puppet-3.7.3/puppet.gemspec
--- puppet-3.7.3_ori/puppet.gemspec	1970-01-01 08:00:00.000000000 +0800
+++ puppet-3.7.3/puppet.gemspec	2014-11-12 14:30:07.000000000 +0800
@@ -0,0 +1,19 @@
+$:.unshift(File.dirname(__FILE__) + '/lib')
+require 'puppet/version'
+
+Gem::Specification.new do |s|
+  s.name = 'puppet'
+  s.version = Puppet::PUPPETVERSION
+  s.platform = Gem::Platform::RUBY
+  s.summary = 'Open source Puppet is a configuration management system'
+  s.description = s.summary
+  s.author = 'Yang Haibo'
+  s.email = 'b40869@freescale.com'
+  s.homepage = 'https://puppetlabs.com/puppet/puppet-open-source'
+  s.license = 'Apache 2.0'
+
+  s.bindir       = 'bin'
+  s.executables  = ['puppet', 'extlookup2hiera']
+  s.require_path = 'lib'
+  s.files = %w(LICENSE README.md Rakefile) + Dir.glob('{lib,spec}/**/*')
+end
