Upstream-Status: Inappropriate [configuration]

Index: horizon-2013.1.2/openstack_dashboard/settings.py
===================================================================
--- horizon-2013.1.2.orig/openstack_dashboard/settings.py
+++ horizon-2013.1.2/openstack_dashboard/settings.py
@@ -29,7 +29,7 @@ warnings.formatwarning = lambda message,
                                 '%s: %s' % (category.__name__, message)
 
 ROOT_PATH = os.path.dirname(os.path.abspath(__file__))
-BIN_DIR = os.path.abspath(os.path.join(ROOT_PATH, '..', 'bin'))
+BIN_DIR = os.path.abspath(os.path.join(ROOT_PATH, 'bin'))
 
 if ROOT_PATH not in sys.path:
     sys.path.append(ROOT_PATH)
