Index: gcc-linaro-4.8-2014.03/gcc/config/aarch64/aarch64-linux.h
===================================================================
--- gcc-linaro-4.8-2014.03.orig/gcc/config/aarch64/aarch64-linux.h
+++ gcc-linaro-4.8-2014.03/gcc/config/aarch64/aarch64-linux.h
@@ -21,10 +21,23 @@
 #ifndef GCC_AARCH64_LINUX_H
 #define GCC_AARCH64_LINUX_H
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}%{mabi=ilp32:_ilp32}.so.1"
-
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
 
+#if TARGET_BIG_ENDIAN_DEFAULT
+#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{!mlittle-endian:_be}.so.1"
+
+#define LINUX_TARGET_LINK_SPEC  "%{h*}		\
+   %{static:-Bstatic}				\
+   %{shared:-shared}				\
+   %{symbolic:-Bsymbolic}			\
+   %{rdynamic:-export-dynamic}			\
+   -dynamic-linker " GNU_USER_DYNAMIC_LINKER "	\
+   -X						\
+   %{mbig-endian:-EB} %{mlittle-endian:-EL}     \
+   -maarch64linux%{mabi=ilp32:32}%{!mlittle-endian:b}"
+#else
+#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}%{mabi=ilp32:_ilp32}.so.1"
+
 #define LINUX_TARGET_LINK_SPEC  "%{h*}		\
    %{static:-Bstatic}				\
    %{shared:-shared}				\
@@ -34,6 +47,7 @@
    -X						\
    %{mbig-endian:-EB} %{mlittle-endian:-EL}     \
    -maarch64linux%{mabi=ilp32:32}%{mbig-endian:b}"
+#endif
 
 #define LINK_SPEC LINUX_TARGET_LINK_SPEC
 
